solvation_analysis/tests/test_solvation_analysis.py

```diff
@@ -77,28 +85,62 @@ def u_real_named(u_real):
 
 @pytest.fixture
 def atom_groups(u_real):
-    li_atoms = u_real.atoms.select
-    pf6_atoms = u_real.atoms.select
-    bn_atoms = u_real.atoms.select
-    fec_atoms = u_real.atoms.select_atoms()
+    li_atoms = u_real.atoms.select_atoms("type 22")
+    pf6_atoms = u_real.atoms.select_atoms("type 20").residues.atoms
+    bn_atoms = u_real.atoms.select_atoms("type 5").residues.atoms
+    fec_atoms = u_real.atoms.select_atoms("type 21").residues.atoms
     atom_groups = {"li": li_atoms, "pf6": pf6_atoms, "bn": bn_atoms, "fec": fec_atoms}
-
-
-def test_solvation_analysis_imported():
-    """Sample test, will always pass so long as import statement worked"""
-    assert "solvation_analysis" in sys.modules
+    return atom_groups
 
 
 def test_get_atom_group(u_real_named):
-    return
-
+    u = u_real_named
+    res_group = u.residues[1:5]
+    atom_group = u.atoms[1:5]
+    res = u.residues[1]
+    atom = u.atoms[1]
+    groups = [res_group, atom_group, res, atom]
+    for group in groups:
+        assert isinstance(get_atom_group(u, group), mda.core.groups.AtomGroup)
+
+
+def test_get_closest_n_mol(u_grid_1, u_real, atom_groups):
+    # TODO: fix grid trajectory
+    # TODO: use grid trajectory to test atom closeness / res picking
+    # u = u_grid_1
+    # test_atom = u.atoms[42]
+    # atoms = get_closest_n_mol(u, test_atom)
+    # assert len(atoms) == 5
+    test_li = atom_groups["li"][0]
+    shell_sizes = range(2, 8)
+    for size in shell_sizes:
+        shell = get_closest_n_mol(u_real, test_li, n_mol=size)
+        assert len(shell.residues) == size + 1
+
+    radii_range = range(0, 5)
+    default_shell, default_resids, default_radii = get_closest_n_mol(
+        u_real, test_li, return_resids=True, return_radii=True
+    )
+    for rad in radii_range:
+        shell, resids, radii = get_closest_n_mol(
+            u_real, test_li, radius=rad, return_resids=True, return_radii=True
+        )
+        assert shell == default_shell
+        np.testing.assert_equal(resids, default_resids)
+        np.testing.assert_equal(radii, default_radii)
```

Comment:
Just a suggestion--`np.assert_allclose()` is a bit safer for floating point comparisons. For `resids` it is safe, those are integers, but the `radii` are floats so there's a tiny little risk that the comparison fails to achieve exact equality.

I think the advice still holds even when you are trying to show that the floating points values really are basically identical/unchanged, those the reasons are sometimes mind-bending even when I read the explanations.

Reply:
Thanks this is super helpful